futures-channel/benches/sync_mpsc.rs

```diff
@@ -108,71 +107,67 @@ struct TestSender {
 // Could be a Future, it doesn't matter
 impl Stream for TestSender {
     type Item = u32;
-    type Error = ();
-
-    fn poll_next(&mut self, cx: &mut task::Context) -> Poll<Option<Self::Item>, Self::Error> {
-        try_ready!(self.tx.poll_ready(cx).map_err(|_| ()));
-        self.tx.start_send(self.last + 1).unwrap();
-        self.last += 1;
-        assert!(self.tx.poll_flush(cx).unwrap().is_ready());
-        Ok(Async::Ready(Some(self.last)))
+
+    fn poll_next(mut self: PinMut<Self>, cx: &mut task::Context)
+        -> Poll<Option<Self::Item>>
+    {
+        let this = &mut *self;
```

Comment:
Why?

Reply:
I need to access `self.tx` and `self.last` in overlapping regions.

Reply:
ah interesting. The `unsafe_unpinned` macros also have a similar problem I think